features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/mobileapp/js/lib/handlebars/custom.handlebars.helpers.js

```diff
@@ -0,0 +1,292 @@
+     *
+     * @param {string} [operator=null] operator of the expression; allowed operators are ==, !=,
+     *     ===, <,
+     *     <=, >, >= and typeof
+     *
+     * @example
+     *          {{compare this.age "18" operator="<="}}
+     *              <p>He is a child</p>
+     *          {{else}}
+     *              <p>He is an adult</p>
+     *          {{/compare}}
+     *
+     *          {{compare item "number" operator="typeof"}}
+     *              <p>This is a number</p>
+     *          {{else}}
+     *              <p>This is not a number</p>
+     *          {{/compare}}
+     *
+     */
+    handlebarsInstance.registerHelper('compare', function (lvalue, rvalue, options) {
+
+        if (arguments.length < 3) {
+            throw new Error("Handlebars Helper 'compare' needs 2 parameters");
+        }
+
+        operator = options.hash.operator || "==";
+
+        var operators = {
+            '==': function (l, r) {
+                return l == r;
+            },
+            '===': function (l, r) {
+                return l === r;
+            },
+            '!=': function (l, r) {
+                return l != r;
+            },
+            '<': function (l, r) {
+                return l < r;
+            },
+            '>': function (l, r) {
+                return l > r;
+            },
+            '<=': function (l, r) {
+                return l <= r;
+            },
+            '>=': function (l, r) {
+                return l >= r;
+            },
+            'typeof': function (l, r) {
+                return typeof l == r;
+            }
+        };
+        if (!operators[operator]) {
+            throw new Error("Handlebars Helper 'compare' doesn't know the operator " + operator);
+        }
+
+        var result = operators[operator](lvalue, rvalue);
+        if (result) {
+            return options.fn(this);
+        } else {
+            return options.inverse(this);
+        }
+
+    });
+
+    /**
+     * Registers 'ifCond' block helper in Handlebars. This helper acts similar to an if-clause that
+     * has a single-expression.
+     *
+     * @param lvalue {Object} left hand side value of the expression
+     * @param operator {string} operator of the expression; allowed operators are ==, !=, ===, !==,
+     *     <, <=, >, >=, &&, || and typeof
+     * @param rvalue {Object} right hand side value of the expression
+     *
+     * @example
+     *          {{ifCond this.age "<=" "18"}}
+     *              <p>He is a child</p>
+     *          {{else}}
+     *              <p>He is an adult</p>
+     *          {{/ifCond}}
+     *
+     *          {{ifCond item "typeof" "number"}}
+     *              <p>This is a number</p>
+     *          {{else}}
+     *              <p>This is not a number</p>
+     *          {{/ifCond}}
+     */
+    handlebarsInstance.registerHelper('ifCond', function (lvalue, operator, rvalue, options) {
```

Comment:
These Handlebars helpers were in the 'theme.js' file. Then they were moved here in order to use them in the client-side. See commit 18d954245926fbb23cefbec6836c734f845dd7b7 for more details.